packages/pricing/src/services/pricing-module.ts

```diff
@@ -247,19 +266,377 @@ export default class PricingModuleService<
+      .flat()
+      .filter((r) => !ruleTypeMap.has(r))
+
+    if (invalidRuleAttributeInputs.length) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `Rule types don't exist for attributes: ${[
+          ...new Set(invalidRuleAttributeInputs),
+        ].join(", ")}`
+      )
+    }
+
+    const priceSetRuleTypesCreate: PricingTypes.CreatePriceSetRuleTypeDTO[] = []
+
+    inputs.forEach((d) => {
+      const priceSet = priceSetMap.get(d.priceSetId)
+
+      for (const r of d.rules) {
+        if (priceSetRuleTypeMap.get(d.priceSetId)!.has(r.attribute)) {
+          continue
+        }
+
+        priceSetRuleTypesCreate.push({
+          rule_type: ruleTypeMap.get(r.attribute) as RuleTypeDTO,
+          price_set: priceSet as unknown as PriceSetDTO,
+        })
       }
+    })
+
+    await this.priceSetRuleTypeService_.create(
+      priceSetRuleTypesCreate as unknown as PricingTypes.CreatePriceSetRuleTypeDTO[],
+      sharedContext
+    )
+  }
+
+  @InjectManager("baseRepository_")
+  async addPrices(
+    data: AddPricesDTO | AddPricesDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<PricingTypes.PriceSetDTO[]> {
+    const input = Array.isArray(data) ? data : [data]
+
+    await this.addPrices_(input, sharedContext)
+
+    return await this.list(
+      { id: input.map((d) => d.priceSetId) },
+      { relations: ["money_amounts"] }
+    )
+  }
+
+  @InjectTransactionManager(shouldForceTransaction, "baseRepository_")
+  protected async addPrices_(
+    input: AddPricesDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ) {
+    const priceSets = await this.list(
+      { id: input.map((d) => d.priceSetId) },
+      { relations: ["rule_types"] },
+      sharedContext
+    )
+
+    const priceSetMap = new Map(priceSets.map((p) => [p.id, p]))
+
+    const ruleTypeMap: Map<string, Map<string, RuleTypeDTO>> = new Map(
+      priceSets.map((priceSet) => [
+        priceSet.id,
+        new Map(
+          priceSet.rule_types?.map((rt) => [rt.rule_attribute, rt]) ?? []
+        ),
+      ])
+    )
+
+    input.forEach(({ priceSetId, prices }) => {
+      const priceSet = priceSetMap.get(priceSetId)
+
+      if (!priceSet) {
+        throw new MedusaError(
+          MedusaError.Types.INVALID_DATA,
+          `Price set with id: ${priceSetId} not found`
+        )
+      }
+
+      const ruleAttributes = prices
+        .map((d) => (d.rules ? Object.keys(d.rules) : []))
+        .flat()
+
+      const invalidRuleAttributes = ruleAttributes.filter(
+        (r) => !ruleTypeMap.get(priceSetId)!.has(r)
+      )
+
+      if (invalidRuleAttributes.length > 0) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_FOUND,
+          `Rule types don't exist for: ${invalidRuleAttributes.join(", ")}`
+        )
+      }
+    })
+
+    await input.forEach(async ({ priceSetId, prices }) => {
+      await Promise.all(
+        prices.map(async (ma) => {
```

Comment:
todo(question): Same as my previous comment, we could bulk create for each entities and then loop on the results to bulk create the rest and so on